core/math/math_funcs.h

```diff
@@ -362,7 +362,7 @@ class Math {
 	}
 
 	static _ALWAYS_INLINE_ bool is_zero_approx(float s) {
-		return abs(s) < CMP_EPSILON;
+		return abs(s) < CMP_EPSILON_32;
```

Comment:
Assuming the above comment is implemented and `CMP_EPSILON_32` is a stricter epsilon value, we can do the same as for 64-bit floats and enable a stricter check behind `PRECISE_MATH_CHECKS`:

```suggestion
#ifdef PRECISE_MATH_CHECKS
		return abs(s) < CMP_EPSILON_64;
#else
		return abs(s) < CMP_EPSILON;
#endif // PRECISE_MATH_CHECKS
```